bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v1/GenericContainerImplTest.java

```diff
@@ -0,0 +1,68 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2019 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v1;
+
+import static org.junit.Assert.*;
+
+import org.apache.sling.api.resource.Resource;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.adobe.cq.export.json.SlingModelFilter;
+import com.adobe.cq.wcm.core.components.context.CoreComponentTestContext;
+import com.adobe.cq.wcm.core.components.models.GenericContainer;
+import com.adobe.cq.wcm.core.components.testing.MockSlingModelFilter;
+
+import io.wcm.testing.mock.aem.junit.AemContext;
+
+public class GenericContainerImplTest {
```

Comment:
Please add more elaborate test cases, such as:
- no background color/image defined
- background color/image defined (in properties) but content policy disables it
- nested containers

Please also test JSON export for all test cases.